apps/gp2-server/test/data-providers/project.data-provider.test.ts

```diff
@@ -112,83 +113,129 @@ describe('Project Data Provider', () => {
-        const email = 'tony@starkenterprises.com';
-        const project = getGraphQLProject();
-        project.flatData.pmEmail = email;
-        const { pmEmail } = parseProjectToDataObject(project);
-        expect(pmEmail).toEqual(email);
-      });
-
-      test('lead emails are added if available', () => {
-        const email = 'tony@starkenterprises.com';
-        const project = getGraphQLProject();
-        project.flatData.leadEmail = email;
-        const { leadEmail } = parseProjectToDataObject(project);
-        expect(leadEmail).toEqual(email);
-      });
-
-      test('description is added if available', () => {
-        const expectedDescription = 'this is a description';
-        const project = getGraphQLProject();
-        project.flatData.description = expectedDescription;
-        const { description } = parseProjectToDataObject(project);
-        expect(description).toEqual(expectedDescription);
-      });
-
-      test('keywords are added', () => {
-        const expectedKeywords = ['Outreach'];
-        const project = getGraphQLProject();
-        project.flatData.keywords = expectedKeywords;
-        const { keywords } = parseProjectToDataObject(project);
-        expect(keywords).toEqual(expectedKeywords);
+      describe('members', () => {
+        test('the members in project are parsed', () => {
+          const Project = getGraphQLProject();
+          Project.flatData.members = [getGraphQLProjectMember()];
+          const { members } = parseProjectToDataObject(Project);
+          expect(members).toEqual([
+            {
+              userId: '42',
+              firstName: 'Tony',
+              lastName: 'Stark',
+            },
+          ]);
+        });
+
+        test('undefined members returns empty array', () => {
+          const Project = getGraphQLProject();
+          Project.flatData.members = undefined!;
+          const { members } = parseProjectToDataObject(Project);
+          expect(members).toEqual([]);
+        });
+        test('avatar urls are added if available', () => {
+          const Project = getGraphQLProject();
+          const member = getGraphQLProjectMember();
+          member!.user![0]!.flatData.avatar = [{ id: 'avatar-id' }];
+          Project.flatData.members = [member];
+          const { members } = parseProjectToDataObject(Project);
+          expect(members[0]?.avatarUrl).toEqual(
+            `${baseUrl}/api/assets/${appName}/avatar-id`,
+          );
+        });
+
+        test('should skip the user from the result if the user property is undefined', () => {
+          const Project = getGraphQLProject();
+          const member = getGraphQLProjectMember();
+          member!.user = undefined!;
+          Project.flatData.members = [member];
+          const { members } = parseProjectToDataObject(Project);
+          expect(members).toEqual([]);
+        });
+        test('pm emails are added if available', () => {
```

Comment:
it would be a lot easier to just expect all the information to be present from the beginning and assert all at once, this just seems a little unnecessary. If these are defaulted to empty strings or different values then you can test those separately

Reply:
The default value from members is an empty array. 

We then test the values later.